addons/point_of_sale/models/pos_session.py

```diff
@@ -1571,11 +1573,56 @@ def _load_model(self, model):
             raise NotImplementedError(_("The function to load %s has not been implemented.", model))
 
     def load_pos_data(self):
-        loaded_models = {}
-        self = self.with_context(loaded_models=loaded_models)
+        loaded_data = {}
+        self = self.with_context(loaded_data=loaded_data)
         for model in self._pos_ui_models_to_load():
-            loaded_models[model] = self._load_model(model)
-        return loaded_models
+            loaded_data[model] = self._load_model(model)
+        self._pos_data_process(loaded_data)
+        return loaded_data
+
+    def _get_attributes_by_ptal_id(self):
+        product_attributes = self.env['product.attribute'].search([('create_variant', '=', 'no_variant')])
```

Comment:
yop, tu trouve pas que c'est super moche? peut-etre on pourrait s'en sortir un peu mieux en sql non?

Reply:
Answering this for history: I simply refactored the old ugly code while keeping (and improving) +/- the same logic without wasting too much time.
In SQL that would may be more readable but it depends if we want to invest time on it

Reply:
Todo that later